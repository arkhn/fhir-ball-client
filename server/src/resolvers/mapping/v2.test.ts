import { Photon } from '@prisma/photon'

import importMappingV2 from './v2'
import * as mappingV2 from '../../../test/fixtures/chimio-mapping-v2.json'

const mockCreateResource = jest.fn()
jest.mock('@prisma/photon', () => ({
  Photon: jest.fn().mockImplementation(() => ({
    resources: {
      create: (data: any) => mockCreateResource(data),
    },
  })),
}))

describe('import mapping V2', () => {
  const sourceId = '01234567'
  const resourceCount = 2
  const { resources } = mappingV2 as any

  beforeEach(() => {
    mockCreateResource.mockClear()
  })

  it('should send a query per resource', async () => {
    await importMappingV2(new Photon(), sourceId, resources)
    expect(mockCreateResource).toHaveBeenCalledTimes(resourceCount)
    expect(mockCreateResource.mock.calls[0]).toMatchSnapshot() // EpisodeOfCare - HopitalStay
    expect(mockCreateResource.mock.calls[1]).toMatchSnapshot() // Patient
  })

  it('should have cleaned the resource and attributes', async () => {
    await importMappingV2(new Photon(), sourceId, resources)
    expect(mockCreateResource.mock.calls[0][0]).toEqual({
      data: {
        label: resources[0].label,
        primaryKeyOwner: resources[0].primaryKeyOwner,
        primaryKeyTable: resources[0].primaryKeyTable,
        primaryKeyColumn: resources[0].primaryKeyColumn,
        source: {
          connect: {
            id: sourceId,
          },
        },
        definitionId: 'EpisodeOfCare',
        attributes: {
          create: expect.arrayContaining([
            {
              path: 'period.start',
              comments: null,
              mergingScript: 'merge_concat',
              inputs: {
                create: expect.any(Array),
              },
            },
          ]),
        },
      },
    })
  })
})
