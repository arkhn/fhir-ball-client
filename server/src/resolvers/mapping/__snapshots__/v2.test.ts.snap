// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import mapping V2 should send a query per resource 1`] = `
Array [
  Object {
    "data": Object {
      "attributes": Object {
        "create": Array [
          Object {
            "comments": Object {
              "create": Object {
                "author": Object {
                  "connect": Object {
                    "email": "admin@arkhn.com",
                  },
                },
                "content": "test",
              },
            },
            "definitionId": "",
            "id": "ck5o0c3400004xzut9wtfoymo",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c3430005xzut626bvgxb",
                  "script": null,
                  "staticValue": "internal",
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.reference",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c3480007xzutrpot4mw4",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c34a0009xzutjn2rrsd7",
                  "script": null,
                  "staticValue": "/Organization/{id}",
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.identifier.system",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c34c0011xzutun7in6e2",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c34g0014xzutsdu9xc3z",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "SERVICE",
                      "id": "ck5o0c34e0012xzutavhpg1gr",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.identifier.value",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c34i0016xzutgm03wu9c",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c34n0020xzuthqycetmu",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ETATSEJ",
                      "id": "ck5o0c34l0018xzuta05qgno6",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "status",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c34p0022xzut4oapfln0",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c34t0026xzutc511n76n",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "SECLEUNIK",
                      "id": "ck5o0c34q0024xzutdivc7jel",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].value",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c34w0028xzutm5yg0oun",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c3500032xzut3h0jz23u",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADMISSION",
                      "id": "ck5o0c34y0030xzut2ap9rs5z",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "id": "ck5o0c3550036xzut2zd6g5dt",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADMISSION2",
                      "id": "ck5o0c3530034xzutfqrt4ykq",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": "select_first_not_empty",
            "path": "identifier[1].value",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c3570038xzutox1k7lue",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c35b0042xzut3vw6tirm",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "TWCLEUNIK",
                      "id": "ck5o0c3590040xzutto1i0sh0",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "type[0].coding[0].code",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c35e0044xzutcvf5p65b",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c35u0055xzut4lhvoo7n",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "LIBELTW",
                      "id": "ck5o0c35h0046xzutwsof9xp2",
                      "joins": Object {
                        "create": Array [
                          Object {
                            "id": "ck5o0c35l0048xzutudu559h5",
                            "tables": Object {
                              "create": Array [
                                Object {
                                  "column": "TWCLEUNIK",
                                  "id": "ck5o0c35o0051xzutzfvbq3pe",
                                  "owner": "OPS$ACHQ1ABC",
                                  "table": "SEJOUR",
                                },
                                Object {
                                  "column": "TWCLEUNIK",
                                  "id": "ck5o0c35r0053xzutkvtmg879",
                                  "owner": "OPS$ACHQ1ABC",
                                  "table": "TYPESEJ",
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "owner": "OPS$ACHQ1ABC",
                      "table": "TYPESEJ",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "type[0].coding[0].display",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c35x0057xzutw5y89297",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c3620061xzutosdnxw7m",
                  "script": "clean_date",
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEOUT",
                      "id": "ck5o0c3600059xzutpjhfg0d8",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "id": "ck5o0c3660065xzutlbhcru1x",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "HEUREOUT",
                      "id": "ck5o0c3640063xzut6i0itybh",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": "merge_concat",
            "path": "period.end",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c3690067xzutbcr73acc",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c36d0071xzutquz0bdc3",
                  "script": "clean_date",
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEIN",
                      "id": "ck5o0c36b0069xzutscaaio2h",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "id": "ck5o0c36h0075xzutmtbz6ljj",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "HEUREIN",
                      "id": "ck5o0c36f0073xzut83oazxki",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": "merge_concat",
            "path": "period.start",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c36k0077xzut5v4reilc",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c36l0079xzutvvf1xe16",
                  "script": null,
                  "staticValue": "internal",
                },
              ],
            },
            "mergingScript": null,
            "path": "patient.reference",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c36o0081xzut0cvjy55s",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c36q0083xzuthi0q4gg9",
                  "script": null,
                  "staticValue": "/Patient/{id}",
                },
              ],
            },
            "mergingScript": null,
            "path": "patient.identifier.system",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c36s0085xzutsxkuiunc",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c36x0089xzutgoo0836z",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "PACLEUNIK",
                      "id": "ck5o0c36v0087xzutv6gwndhg",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "patient.identifier.value",
          },
        ],
      },
      "definitionId": "EpisodeOfCare",
      "id": "ck5o0c33r0002xzutgasxa0ca",
      "label": "Hospital Stay",
      "primaryKeyColumn": "SECLEUNIK",
      "primaryKeyOwner": "OPS$ACHQ1ABC",
      "primaryKeyTable": "SEJOUR",
      "source": Object {
        "connect": Object {
          "id": "01234567",
        },
      },
    },
  },
]
`;

exports[`import mapping V2 should send a query per resource 2`] = `
Array [
  Object {
    "data": Object {
      "attributes": Object {
        "create": Array [
          Object {
            "definitionId": "",
            "id": "ck5o0c3460006xzuti581wmzv",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c34c0010xzutu6vjy417",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "INACTIVPAT",
                      "id": "ck5o0c3490008xzut0mvunkey",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "active",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c34f0013xzutraa4nyy1",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c34j0017xzut58opiyr0",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADR1PAT",
                      "id": "ck5o0c34h0015xzutb1aaf5xo",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "id": "ck5o0c34o0021xzuta7v4vzc2",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADR2PAT",
                      "id": "ck5o0c34l0019xzutzwc9ehzp",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "id": "ck5o0c34t0025xzutenfeuylz",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADR3PAT",
                      "id": "ck5o0c34q0023xzutwy4nuh8s",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": "merge_concat",
            "path": "address[0].line[0]",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c34v0027xzut8szefrbv",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c3500031xzutv52lvkts",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "VILLEPAT",
                      "id": "ck5o0c34x0029xzut5lbyujt2",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "address[0].city",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c3520033xzut7j36tpun",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c3560037xzutrxf5zl8d",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "CPOSTPAT",
                      "id": "ck5o0c3540035xzuthkq6fhze",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "address[0].postalCode",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c3580039xzutfupmloig",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c35d0043xzutrx8lbtfi",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "INACTIVPAT",
                      "id": "ck5o0c35b0041xzutqvdedrgz",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "id": "ck5o0c35j0047xzuttycv3b8p",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "IDMOTIFINAC",
                      "id": "ck5o0c35g0045xzutdr4stnjt",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "deceasedBoolean",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c35l0049xzutq46qdgc0",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c35p0052xzutu394risb",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEINACPAT",
                      "id": "ck5o0c35n0050xzutgyh0eohy",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "deceasedDateTime",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c35r0054xzutroefmylo",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c35x0058xzutvrggaij9",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "TELPAT",
                      "id": "ck5o0c35v0056xzutfop5m915",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "telecom[0].value",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c3600060xzutdvjjjopd",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c3630062xzutj4ireatd",
                  "script": null,
                  "staticValue": "official",
                },
              ],
            },
            "mergingScript": null,
            "path": "name[0].use",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c3650064xzut2rj4jh2b",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c3690068xzutqmxzcd6j",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "NOMPAT",
                      "id": "ck5o0c3670066xzutw5g97cnl",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "name[0].family",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c36b0070xzutomr8h68j",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c36g0074xzutrdjmgkfn",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "PRENOMPAT",
                      "id": "ck5o0c36d0072xzutgzo2y62d",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "name[0].given[0]",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c36j0076xzut05nkluin",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c36l0078xzut68kj000t",
                  "script": null,
                  "staticValue": "maiden",
                },
              ],
            },
            "mergingScript": null,
            "path": "name[1].use",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c36n0080xzutrpvzixlf",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c36r0084xzut9q8ei4ga",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "NOMJFPAT",
                      "id": "ck5o0c36p0082xzutsm8qbban",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "name[1].family",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c36u0086xzutlndd8xgi",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c36z0090xzutbxpqwcns",
                  "script": "clean_date",
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATENAISPA",
                      "id": "ck5o0c36w0088xzutjj30g1op",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "birthDate",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c3710091xzutmzwekpjn",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c3760093xzut40uvrgtm",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "SEXEPAT",
                      "id": "ck5o0c3740092xzutj6s398fo",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "gender",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c3780094xzut77nv9ox7",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c3790095xzut4g7vnget",
                  "script": null,
                  "staticValue": "http://terminology.interopsante.org/CodeSystem/v2-0203",
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].type.coding[0].system",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c37b0096xzutv6y539hn",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c37d0097xzutsnmh99fh",
                  "script": null,
                  "staticValue": "PI",
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].type.coding[0].code",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c37f0098xzutfj8xl0iv",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c37g0099xzut8joep3zp",
                  "script": null,
                  "staticValue": "Patient internal identifier",
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].type.coding[0].display",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c37i0100xzutcqkdz3kw",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c37k0101xzuty2yxs9kx",
                  "script": null,
                  "staticValue": "false",
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].type.coding[0].userSelected",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c37l0102xzuteqz4cxh8",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c37p0104xzut6xai9nc9",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "NOOBSPAT",
                      "id": "ck5o0c37n0103xzutg2xkk2kj",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].value",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c37r0105xzutz7dm9hia",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c37v0107xzut1b4d9csu",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "PACLEUNIK",
                      "id": "ck5o0c37t0106xzutgef9ixg7",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[1].value",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c37x0108xzutsvrhnwhe",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c3830110xzutcfxcegmb",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DOSSIERPAT",
                      "id": "ck5o0c3810109xzutrclm013e",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[2].value",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c3850111xzuts71ow2kl",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c3890113xzut51brcly0",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEINPAT",
                      "id": "ck5o0c3870112xzut4ktuv4ol",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[2].period.start",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c38b0114xzut16bnfmd3",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c38h0116xzute4pgahou",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEOUTPAT",
                      "id": "ck5o0c38d0115xzuthjfqgmln",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[2].period.end",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c38k0117xzutlht8snvf",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c38m0118xzutbnsm0u4k",
                  "script": null,
                  "staticValue": "/Organization",
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.type",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c38o0119xzutrh3t3fd9",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c38q0120xzutfae1319f",
                  "script": null,
                  "staticValue": "/Organization/{id}",
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.identifier.system",
          },
          Object {
            "definitionId": "",
            "id": "ck5o0c38s0121xzut09m5pnem",
            "inputs": Object {
              "create": Array [
                Object {
                  "id": "ck5o0c38v0123xzutxwy8q9lo",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "CODESERV",
                      "id": "ck5o0c38t0122xzutw260z8r0",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.identifier.value",
          },
        ],
      },
      "definitionId": "Patient",
      "id": "ck5o0c33r0003xzut4i320wg2",
      "label": "",
      "primaryKeyColumn": "PACLEUNIK",
      "primaryKeyOwner": "OPS$ACHQ1ABC",
      "primaryKeyTable": "PATIENT",
      "source": Object {
        "connect": Object {
          "id": "01234567",
        },
      },
    },
  },
]
`;
